### PR TITLE
[CCR] Only auto follow indices when all primary shards have started

### LINKED ISSUE #35480: [CI] AutoFollowIT#testConflictingPatterns failing
Hey all, not sure if this is real or not, but popped up on my PR test. May be worth investigating?

CI: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request/2600/

```
./gradlew :x-pack:plugin:ccr:internalClusterTest -Dtests.seed=4205CBCCCD58E520 -Dtests.class=org.elasticsearch.xpack.ccr.AutoFollowIT -Dtests.method="testConflictingPatterns" -Dtests.security.manager=true -Dtests.locale=es-PY -Dtests.timezone=Pacific/Majuro -Dcompiler.java=11 -Druntime.java=8
```

stacks

```
  1> Caused by: org.elasticsearch.transport.RemoteTransportException: [leader0][127.0.0.1:53885][indices:data/read/xpack/ccr/shard_changes[s]]
  1> Caused by: org.elasticsearch.index.shard.IndexShardClosedException: CurrentState[CLOSED] Closed
  1> 	at org.elasticsearch.index.shard.GlobalCheckpointListeners.close(GlobalCheckpointListeners.java:158) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:103) ~[elasticsearch-core-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.core.internal.io.IOUtils.close(IOUtils.java:61) ~[elasticsearch-core-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.index.shard.IndexShard.close(IndexShard.java:1187) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.index.IndexService.closeShard(IndexService.java:436) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.index.IndexService.removeShard(IndexService.java:419) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.index.IndexService.close(IndexService.java:277) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.indices.IndicesService.removeIndex(IndicesService.java:638) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
  1> 	at org.elasticsearch.indices.cluster.IndicesClusterStateService.deleteIndices(IndicesClusterStateService.java:286) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
```

### PR DESCRIPTION
**Pull request by martijnvg**
This change adds an extra check that verifies that all primary shards
have been started of an index that is about to be auto followed.

If not all primary shards have been started for an index
then the next auto follow run will try to follow to auto follow
this index again.

Closes #35480

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java [hunk 1]
```diff
@@ -367,7 +369,10 @@ private void finalise(int slot, AutoFollowResult result) {
             List<Index> leaderIndicesToFollow = new ArrayList<>();
             for (IndexMetaData leaderIndexMetaData : leaderClusterState.getMetaData()) {
                 if (autoFollowPattern.match(leaderIndexMetaData.getIndex().getName())) {
-                    if (followedIndexUUIDs.contains(leaderIndexMetaData.getIndex().getUUID()) == false) {
+                    IndexRoutingTable indexRoutingTable = leaderClusterState.routingTable().index(leaderIndexMetaData.getIndex());
+                    if (indexRoutingTable != null &&
+                        indexRoutingTable.allPrimaryShardsActive() &&
```
Comment (jasontedor) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java]: Looks good. Can you leave a comment explaining this though?

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by martijnvg** [comment]
I will wait with backporting to the 6.5 branch, when 6.5.2 has been released.

**Comment by martijnvg** [comment]
Thanks @jasontedor for checking. I've backported the bug fix to the 6.5 branch.

**Pull request merged**